packages/vitest/src/node/workspace/resolveWorkspace.ts

```diff
@@ -127,9 +128,141 @@ export async function resolveWorkspace(
+    }
+
+    const newName = originalName
+      ? `${originalName} (${firstConfig.browser})`
+      : firstConfig.browser
+    if (names.has(newName)) {
+      throw new Error(`Cannot redefine the project name for a nameless project. The project name "${firstConfig.browser}" was already defined. All projects in a workspace should have unique names. Make sure your configuration is correct.`)
+    }
+    names.add(newName)
+
+    if (project.config.browser.providerOptions) {
+      vitest.logger.warn(
+        withLabel('yellow', 'Vitest', `"providerOptions"${originalName ? ` in "${originalName}" project` : ''} is ignored because it's overriden by the configs. To hide this warning, remove the "providerOptions" property from the browser configuration.`),
+      )
+    }
+
+    restConfigs.forEach((config, index) => {
+      const browser = config.browser
+      if (!browser) {
+        const nth = index + 1
+        const ending = nth === 2 ? 'nd' : nth === 3 ? 'rd' : 'th'
+        throw new Error(`The browser configuration must have a "browser" property. The ${nth}${ending} item in "browser.instances" doesn't have it. Make sure your${originalName ? ` "${originalName}"` : ''} configuration is correct.`)
+      }
+      const name = config.name
+      const newName = name || (originalName ? `${originalName} (${browser})` : browser)
+      if (names.has(newName)) {
+        throw new Error(
+          [
+            `Cannot define a nested project for a ${browser} browser. The project name "${newName}" was already defined. `,
+            'If you have multiple instances for the same browser, make sure to define a custom "name". ',
+            'All projects in a workspace should have unique names. Make sure your configuration is correct.',
+          ].join(''),
+        )
+      }
+      names.add(newName)
+      const clonedConfig = cloneConfig(project, config)
+      clonedConfig.name = newName
+      newConfigs.push([project, clonedConfig])
+    })
+
+    Object.assign(project.config, cloneConfig(project, firstConfig))
+    project.config.name = newName
+  })
+  newConfigs.forEach(([project, clonedConfig]) => {
+    const clone = TestProject._cloneBrowserProject(project, clonedConfig)
+    resolvedProjects.push(clone)
+  })
+
+  const headedBrowserProjects = resolvedProjects.filter((project) => {
+    return project.config.browser.enabled && !project.config.browser.headless
+  })
+  if (headedBrowserProjects.length > 1) {
+    const message = [
+      `Found multiple projects that run browser tests in headed mode: "${headedBrowserProjects.map(p => p.name).join('", "')}".`,
+      ` Vitest cannot run multiple headed browsers at the same time.`,
+    ].join('')
+    if (!isTTY) {
+      throw new Error(`${message} Please, filter projects with --browser=name or --project=name flag or run tests with "headless: true" option.`)
```

Comment:
Does `--project` filter work? In `test/browser`, I get a prompt like this:

```sh
$ pnpm -C test/browser test-fixtures --project=chromium

> @vitest/test-browser@ test-fixtures /home/hiroshi/code/others/vitest/test/browser
> vitest "basic" "--project=chromium"

? Found multiple projects that run browser tests in headed mode: "chromium", "firefox", "webkit". Vitest cannot run multiple headed browsers at the same time. Select a single project to run or cancel and run tests with "headless: true" option. Note that you can also start tests with --browser=name or --project=name flag. › 
```

Reply:
Fixed